synapse/handlers/deactivate_account.py

```diff
@@ -261,11 +263,40 @@ async def _part_user(self, user_id: str) -> None:
         user = UserID.from_string(user_id)
 
         rooms_for_user = await self.store.get_rooms_for_user(user_id)
+        requester = create_requester(user, authenticated_entity=self._server_name)
+        should_erase = await self.store.is_user_erased(user_id)
+
         for room_id in rooms_for_user:
-            logger.info("User parter parting %r from %r", user_id, room_id)
             try:
+                # Before parting the user, redact all membership events if requested
+                if should_erase:
+                    event_ids = await self.store.get_membership_event_ids_for_user(
+                        user_id, room_id
+                    )
+                    events: list[EventBase] = await self.store.get_events_as_list(
+                        event_ids
+                    )
+                    for event in events:
+                        has_profile = (
+                            "displayname" in event.content
+                            or "avatar_url" in event.content
+                        )
+                        if has_profile and "redacted_because" not in event.unsigned:
```

Comment:
```suggestion
                        if has_profile:
```

if the content has profile information then it isn't redacted